### PR TITLE
Clarify error for AssemblyVersion with wildcard when deterministic

### PR DESCRIPTION
**Pull request by jcouv**
Fixes https://github.com/dotnet/roslyn/issues/22660

### CODE REVIEW

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for review. Small tweak to diagnostics. Thanks

File: src/Compilers/CSharp/Portable/CSharpResources.resx [hunk 1]
```diff
@@ -3900,8 +3900,11 @@ You should consider suppressing the warning only if you're sure that you don't w
   <data name="ERR_InvalidVersionFormat" xml:space="preserve">
     <value>The specified version string does not conform to the required format - major[.minor[.build[.revision]]]</value>
   </data>
+  <data name="ERR_InvalidVersionFormatDeterministic" xml:space="preserve">
+    <value>The specified version string does not conform to the required format - major[.minor[.build[.revision]]] (without wildcards, due to determinism)</value>
```
Comment (agocke) [inline src/Compilers/CSharp/Portable/CSharpResources.resx]: This message is still really hard to understand. I would prefer something like "The specified version string contains wildcards, which are not compatible with determinism. Either remove wildcards from the version string, or disable determinism for this compilation".

Comment (jcouv) [reply src/Compilers/CSharp/Portable/CSharpResources.resx]: The problem is the diagnostic can be reached either if you have wildcards, or you did not follow the format. 
I could probably add a an check for presence of wildcard. I'll try that.

File: src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Assembly.cs [hunk 2]
```diff
@@ -107,6 +107,15 @@ public void VersionAttributeErr()
...
                 Diagnostic(ErrorCode.ERR_InvalidVersionFormat, @"""-1""").WithLocation(1, 46));
+
+            string s3 = @"[assembly: System.Reflection.AssemblyVersion(""1.1.1.*"")]";
+
+            var comp3 = CreateStandardCompilation(s3, options: TestOptions.ReleaseDll.WithDeterministic(true));
+            comp3.VerifyDiagnostics(
+                // (1,46): error CS8357: The specified version string contains wildcards, which are not compatible with determinism. Either remove wildcards from the version string, or disable determinism for this compilation
+                // [assembly: System.Reflection.AssemblyVersion("1.1.1.*")]
+                Diagnostic(ErrorCode.ERR_InvalidVersionFormatDeterministic, @"""1.1.1.*""").WithLocation(1, 46)
+                );
```
Comment (cston) [inline src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Assembly.cs]: Consider moving `s3` to a new test.

**Comment by jcouv** [comment]
@agocke The last commit addresses your feedback. Thanks

**Pull request merged**